content/languages/lambdacalc/lc-test.md

```diff
@@ -0,0 +1,114 @@
+
+Please refer to specific documentation for further information on the modules which `lc-test` utilizes.
+- [Chai](chai-docs) for assertions
+- [Lambda Calculus](lc-docs) for compilation
+
+## Basic Usage
+
+### Example tests
+```javascript
+import { assert, config, LC, getSolution } from "./lc-test.js";
+
+LC.configure({purity: "Let", numEncoding: "Church"});
+const { multiply } = LC.compile(getSolution());
+
+describe("Fixed tests", () => {
+  it("can multiply church numbers", () => {
+    assert.numEql(multiply(5)(0), 0);
+    assert.numEql(multiply(1)(8), 8);
+    assert.numEql(multiply(7)(9), 63);
+  });
+});
+```
+
+### Components
+
+`lc-test` provides the following utilities:
+* `chai.assert` as `assert`
+* `chai.config` as `config`
+* `LC = { compile, configure, fromInt, toInt }`
+* `getSolution`, `getPreloaded` and `getSolutionWithPreloaded`
+
+#### `assert`
+
+`assert` is the standard `chai.assert`, extended with the following the following additional assertion:
+
+**`assert.numEql(actual, expected [ , message ] )`**
+
+`actual` and `expected` are automatically decoded according to the current `numEncoding`, if necessary, before making an equality assertion between them. A custom failure message can optionally be set with `message`.
+
+#### `config`
+
+`config` is merely a re-export of `chai.config`. A notable use case is allowing arrays to be displayed fully in assertion messages, using `config.truncateThreshold = 0`.
+
+#### `getSolution` / `getPreloaded` / `getSolutionWithPreloaded`
+`getSolution()` returns the user `Solution` code, `getPreloaded` returns the author `Preloaded` code, `getSolutionWithPreloaded()` returns a safe version of `Preloaded` and `Solution` code, concatenated.
+
+#### `LC.compile`
+
+`LC.compile(getSolution())` returns a JavaScript `Object` with all top level definitions from the compiled Lambda Calculus source code. Each definition is a `Function` which can be directly called.
+
+:::note
+Use destructuring to conveniently extract the desired definitions, renaming them to valid Javascript identifiers.
+
+```javascript
+const solution = LC.compile(getSolution());
+const { "is-empty": isEmpty } = solution;
+```
+
+Note that definitions in Lambda Calculus source code can be overwritten; for each name defined multiple times, only the _last_ definition will be exported.
+:::
+
+#### `LC.configure`
+
+Default configuration is
+
+```javascript
+{ verbosity: "Calm"    //  Calm | Concise | Loquacious | Verbose
+, purity: "PureLC"     //  Let | LetRec | PureLC
+, numEncoding: "None"  //  None | Church | Scott | BinaryScott
+}
+```
+
+`LC.configure();` can be used to change the settings, by passing an object with key-values of the settings to be changed. For example `LC.configure({purity: "Let", numEncoding: "Scott"})`
```

Comment:
Sorry.

I intended "`LC.configure` can be used .. For example `LC.configure({ purity: "Let", numEncoding: "Scott" });`".

The first `configure` refers to the function _name,_ so should not have `()` or `;`. The second refers to an actual invocation.

Reply:
Ah ok